storage/store.go

```diff
@@ -341,6 +338,18 @@ func (s *Store) String() string {
 	return fmt.Sprintf("store=%d:%d (%s)", s.Ident.NodeID, s.Ident.StoreID, s.engine)
 }
 
+// context returns a base context to pass along with commands being executed.
+// TODO(tschottdorf): currently we initialize the context here, but for client
+// requests we would ideally want one from higher up:
+// node -> local sender -> store
+// so that we can log relevant information such as the client address.
+// that however means passing a context through the local sender.
+func (s *Store) context() log.Context {
```

Comment:
If we're going to pass through a context, why make it logging-specific? I'd rather create a more generalized context which has a log context as part of it. That way we can add additional items (such as explicit tracing, etc).